docs/manuals/gui/extension/dynamic_element/index.md

```diff
@@ -0,0 +1,329 @@
+
+Here are the key things to know when creating a React component that implements the
+front-end for a custom visual element:
+
+- The component name should be one used when the element is declared, using the
+  *react_component* argument of the [`Element` constructor](Element.__init__()^).<br/>
+  If this parameter is not used, Taipy GUI uses a camel case transformation of the element
+  name with a lowercase initial letter.
+- All component source files are bundled in a JavaScript library. We are using
+  [*webpack*](https://webpack.js.org/) for this purpose.
+- Components must be exported by the JavaScript library with the same name as the one
+  used in the [`Element` constructor](Element.__init__()^).
+- Property names must be valid Python identifiers.
+- The camel case transformation of the property names, with a lowercase initial letter, are
+  used to name the keys of the *props* argument for the functional component.<br/>
+  We recommend that we declare a TypeScript interface that provides better typing
+  for this argument: each property of the interface should have the transformed name
+  of the element property.<br/>
+  Note that you can overwrite the name of the property in the React component by using
+  the *js_name* argument of the [`ElementProperty` constructor](ElementProperty.__init__()^).
+- Dynamic properties use two keys in the functional component *props* argument:
+
+    - The (transformed) property name as described above;
+    - A similar camel case-transformed version of the property name as if it had been
+      prefixed by 'default_'.<br/>
+      This slot in the *props* is used to render the component before the Taipy GUI back-end
+      has submitted any update (which is then handled by the other slot).
+
+To illustrate this, let's assume we want to create a dynamic custom element called
+*sizable_label*. This element has two dynamic properties:
+
+- *label*, the default property, stores a string that the element should display;
+- *size* is a numeric property that stores how large the label should be rendered.
+
+The element declaration would look like this in the definition of the dictionary that
+holds all the element declarations:
+
+```py
+"sizable_label": Element("label", {
+                   "label": ElementProperty(PropertyType.dynamic_string),
+                   "size": ElementProperty(PropertyType.dynamic_number)
+                 }
+```
+
+The default name for the React component would be "SizableLabel", and its properties
+will be named after the element's property names.<br/>
+Here is how the definition for the React component will look like (typically, this code
+will be stored in a file called "SizableLabel.tsx", in the directory
+"&lt;project_dir&gt;/&lt;package&gt;/front-end/src"):
+
+```ts
+interface SizableLabelProps {
+    label: string;
+    defaultLabel?: string;
+    size?: number;
+    defaultSize?: number;
```

Comment:
label?: string;
defaultLabel; string;
size?: number;
defaultSize: number;

Reply:
Oookkk. That shows there are things I don't get yet.
Thanks